lib/handlers/marshalling/route-marshaller.ts

```diff
@@ -73,17 +101,26 @@ export class RouteMarshaller {
       case Protocol.V3:
         const v3Route = marshalledRoute as MarshalledV3Route
         return new V3Route(
-          v3Route.pools.map((marshalledPool) => PoolMarshaller.unmarshal(marshalledPool)),
+          v3Route.pools.map((marshalledPool) => V3PoolMarshaller.unmarshal(marshalledPool)),
           TokenMarshaller.unmarshal(v3Route.input),
           TokenMarshaller.unmarshal(v3Route.output)
         )
+      case Protocol.V4:
+        const v4Route = marshalledRoute as MarshalledV4Route
+        return new V4Route(
+          v4Route.pools.map((marshalledPool) => V4PoolMarshaller.unmarshal(marshalledPool)),
+          TokenMarshaller.unmarshal(v4Route.input),
+          TokenMarshaller.unmarshal(v4Route.output)
+        )
       case Protocol.MIXED:
         const mixedRoute = marshalledRoute as MarshalledMixedRoute
         const tpools = mixedRoute.pools.map((tpool) => {
           if (tpool.protocol === Protocol.V2) {
             return PairMarshaller.unmarshal(tpool as MarshalledPair)
+          } else if (tpool.protocol === Protocol.V3) {
+            return V3PoolMarshaller.unmarshal(tpool as V3MarshalledPool)
           } else {
```

Comment:
i would an explicit check for V4 here, and throw exception on else, just like you did in other places